vm-validator/src/vm_validator.rs

```diff
@@ -225,10 +230,34 @@ impl<S: StateView> ModuleCodeBuilder for ValidationState<S> {
             Some(bytes) => bytes,
             None => return Ok(None),
         };
-        let compiled_module = self
+        let override_option_module = self
+            .environment
+            .runtime_environment()
+            .vm_config()
+            .override_option_module;
+        let enable_enum_option = self
+            .environment
             .runtime_environment()
-            .deserialize_into_compiled_module(state_value.bytes())?;
-        let extension = Arc::new(AptosModuleExtension::new(state_value));
+            .vm_config()
+            .enable_enum_option;
+        let (compiled_module, extension) = if override_option_module
+            && enable_enum_option
+            && key.address() == &AccountAddress::ONE
+            && *key.name() == *Identifier::from(ident_str!("option"))
+        {
+            let bytes = Bytes::from(OPTION_MODULE_BYTES.to_vec());
+            let compiled_module = self
+                .runtime_environment()
+                .deserialize_into_compiled_module(&bytes)?;
+            let extension = Arc::new(AptosModuleExtension::new_from_bytes(bytes));
+            (compiled_module, extension)
+        } else {
+            let compiled_module = self
+                .runtime_environment()
+                .deserialize_into_compiled_module(state_value.bytes())?;
+            let extension = Arc::new(AptosModuleExtension::new(state_value));
```

Comment:
Maybw we can have new take runtime_environment and state value, and output compiled module + extension? This way we encapsulate that hash is computed over new module bytes, and do overrides inside as well?

Reply:
updated the code to use `set_bytes` to override bytes in state value